pkg/apis/v1/nodeclaim_conversion.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/apis"
+
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+	"sigs.k8s.io/karpenter/pkg/operator/injection"
+)
+
+func (in *NodeClaim) ConvertTo(ctx context.Context, to apis.Convertible) error {
+	v1beta1NC := to.(*v1beta1.NodeClaim)
+	v1beta1NC.Name = in.Name
+	v1beta1NC.UID = in.UID
+	v1beta1NC.Labels = in.Labels
+	v1beta1NC.Annotations = in.Annotations
+
+	// Updating the drift hash on the nodeclaim
+	v1OwnerNodePool, err := in.getV1NodePoolDriftHash(ctx)
+	if err != nil {
+		return err
+	}
+	v1beta1NC.Annotations = lo.Assign(v1beta1NC.Annotations, map[string]string{
+		NodePoolHashAnnotationKey:        v1OwnerNodePool.Annotations[NodePoolHashAnnotationKey],
+		NodePoolHashVersionAnnotationKey: v1OwnerNodePool.Annotations[NodePoolHashVersionAnnotationKey],
+	})
+
+	err = in.Spec.convertTo(ctx, &v1beta1NC.Spec, in.Annotations[V1Beta1KubeletConfiguration])
```

Comment:
How does the kubelet configurationt annotation work when someone changes their reference to the EC2NodeClass that they are using on the v1 API? When we convert back, aren't we going to be converting back to the old kubelet config? Do we have a good solve for this?

Reply:
Will be added in a separate PR